src/QMCHamiltonians/SOECPComponent.cpp

```diff
@@ -462,38 +463,48 @@ SOECPComponent::RealType SOECPComponent::evaluateValueAndDerivatives(ParticleSet
       W.acceptMove(iel);
     }
 
-  ComplexType pairpot;
-  for (int iq = 0; iq < total_knots_; iq++)
-  {
-    ComplexType lsum;
-    for (int il = 0; il < nchannel_; il++)
-    {
-      int l = il + 1;
-      ComplexType msums;
-      for (int m1 = -l; m1 <= l; m1++)
-      {
-        ComplexType Y = sphericalHarmonic(l, m1, dr);
-        for (int m2 = -l; m2 <= l; m2++)
-        {
-          ComplexType ldots;
-          for (int id = 0; id < 3; id++)
-            ldots += lmMatrixElements(l, m1, m2, id) * sMatrixElements(W.spins[iel], W.spins[iel] + deltaS_[iq], id);
-          ComplexType cY = std::conj(sphericalHarmonic(l, m2, rrotsgrid_m_[iq % nknot_]));
-          msums += Y * cY * ldots;
-        }
-      }
-      lsum += sopp_m_[il]->splint(r) * msums;
-    }
-    wvec_[iq] = lsum * psiratio_[iq] * spin_quad_weights_[iq];
-    pairpot += wvec_[iq];
-  }
-
+  ComplexType pairpot = calculateProjector(r, dr, sold);
+  //calculateProjector stores quad points in wvec_
   BLAS::gemv('N', num_vars, total_knots_, 1.0, dratio_.data(), num_vars, wvec_.data(), 1, 1.0, dhpsioverpsi.data(), 1);
 
   return std::real(pairpot);
 #endif
 }
 
+SOECPComponent::RealType SOECPComponent::evaluateValueAndDerivativesExactSpinIntegration(
+    ParticleSet& W,
+    int iat,
+    TrialWaveFunction& Psi,
+    int iel,
+    RealType r,
+    const PosType& dr,
+    const opt_variables_type& optvars,
+    const Vector<ValueType>& dlogpsi,
+    Vector<ValueType>& dhpsioverpsi)
+{
+#ifndef QMC_COMPLEX
+  throw std::runtime_error("SOECPComponent::evaluateValueAndDerivatives should not be called in real build\n");
```

Comment:
Need to deal with this in separate PRs. If SOECP only makes sense in complex builds, we should avoid compiling the class completely in real builds and thus we don't need this ifdef.

Reply:
anything related to spinors/spin-orbit uses complex wave functions and complex spin functions. So that would be a nice change